diff --git a/build/android.ini b/build/android.ini
index ed6a0d0..e7dba3f 100644
--- a/build/android.ini
+++ b/build/android.ini
@@ -14,7 +14,7 @@ cfg_mips_alias_folder_name="mips"
 cfg_default_build_stl="gnu"  #clang is another option
 
 #build machine & build host
-cfg_build_machine="x86_64-apple-darwin14"
+cfg_build_machine="x86_64-unknown-linux-gnu"
 cfg_x86_host_machine="i686-linux-android"
 cfg_arm64_host_machine="aarch64-linux-android"
 cfg_mips_host_machine="mipsel-linux-android"
@@ -24,11 +24,11 @@ cfg_armv7_host_machine="arm-linux-androideabi"
 #only cross compile should define thse variables
 cfg_is_cross_compile=yes
 
-cfg_x86_toolchain_bin="${ANDROID_NDK}/toolchains/x86-${cfg_default_gcc_version}/prebuilt/darwin-x86_64/bin"
-cfg_arm64_toolchain_bin="${ANDROID_NDK}/toolchains/aarch64-linux-android-${cfg_default_gcc_version}/prebuilt/darwin-x86_64/bin"
-cfg_mips_toolchain_bin="${ANDROID_NDK}/toolchains/mipsel-linux-android-${cfg_default_gcc_version}/prebuilt/darwin-x86_64/bin"
-cfg_arm_toolchain_bin="${ANDROID_NDK}/toolchains/arm-linux-androideabi-${cfg_default_gcc_version}/prebuilt/darwin-x86_64/bin"
-cfg_armv7_toolchain_bin="${ANDROID_NDK}/toolchains/arm-linux-androideabi-${cfg_default_gcc_version}/prebuilt/darwin-x86_64/bin"
+cfg_x86_toolchain_bin="${ANDROID_NDK}/toolchains/x86-${cfg_default_gcc_version}/prebuilt/linux-x86_64/bin"
+cfg_arm64_toolchain_bin="${ANDROID_NDK}/toolchains/aarch64-linux-android-${cfg_default_gcc_version}/prebuilt/linux-x86_64/bin"
+cfg_mips_toolchain_bin="${ANDROID_NDK}/toolchains/mipsel-linux-android-${cfg_default_gcc_version}/prebuilt/linux-x86_64/bin"
+cfg_arm_toolchain_bin="${ANDROID_NDK}/toolchains/arm-linux-androideabi-${cfg_default_gcc_version}/prebuilt/linux-x86_64/bin"
+cfg_armv7_toolchain_bin="${ANDROID_NDK}/toolchains/arm-linux-androideabi-${cfg_default_gcc_version}/prebuilt/linux-x86_64/bin"
 
 
 #build arches and build libraries
@@ -49,4 +49,3 @@ cfg_freetype_arm_use_mthumb=no
 #debug and release mode flag
 cfg_build_release_mode="-O3 -DNDEBUG"
 cfg_build_debug_mode="-O0 -g -DDEBUG"
-
diff --git a/build/build.sh b/build/build.sh
index fdf2da1..82b5636 100755
--- a/build/build.sh
+++ b/build/build.sh
@@ -1,4 +1,5 @@
 #!/bin/bash
+set -e
 
 source `pwd`/main.ini
 #
diff --git a/contrib/src/curl/android.patch b/contrib/src/curl/android.patch
index 11bf3e0..ce807e5 100644
--- a/contrib/src/curl/android.patch
+++ b/contrib/src/curl/android.patch
@@ -6,20 +6,20 @@ index 6acb5e4..e795bf3 100644
  /* Define to 1 if the getaddrinfo function is threadsafe. */
  #define HAVE_GETADDRINFO_THREADSAFE 1
  
--/* Define to 1 if you have the `geteuid' function. */
+-/* Define to 1 if you have the 'geteuid' function. */
 -#define HAVE_GETEUID 1
  
  /* Define to 1 if you have the gethostbyaddr function. */
  #define HAVE_GETHOSTBYADDR 1
 @@ -250,11 +248,6 @@
- /* Define to 1 if you have the `getprotobyname' function. */
+ /* Define to 1 if you have the 'getprotobyname' function. */
  #define HAVE_GETPROTOBYNAME 1
  
--/* Define to 1 if you have the `getpwuid' function. */
+-/* Define to 1 if you have the 'getpwuid' function. */
 -#define HAVE_GETPWUID 1
 -
--/* Define to 1 if you have the `getpwuid_r' function. */
+-/* Define to 1 if you have the 'getpwuid_r' function. */
 -#define HAVE_GETPWUID_R 1
  
- /* Define to 1 if you have the `getrlimit' function. */
+ /* Define to 1 if you have the 'getrlimit' function. */
  #define HAVE_GETRLIMIT 1
diff --git a/contrib/src/jpeg/SHA512SUMS b/contrib/src/jpeg/SHA512SUMS
index b2a04dd..9fa73fc 100644
--- a/contrib/src/jpeg/SHA512SUMS
+++ b/contrib/src/jpeg/SHA512SUMS
@@ -1 +1 @@
-c6a2fa316fad8230ffd137469643276f1fef52667536712eae69b24afeea283415e8182a284c0457b54c6e49d6e93cf90d7ace1f2fed129c1e7f64b51c699c29  jpegsrc.v9.tar.gz
+21b8ca3a47162945ba22a979c48e8b54622e07251e9cf802c527095b48996f4db5952e49edaf28c0d8941c76ab2e1b11e50a7a3a41010d56eddf7d03ea031006  jpegsrc.v9.tar.gz
diff --git a/contrib/src/main.mak b/contrib/src/main.mak
index a2a5fe2..18a8e5a 100644
--- a/contrib/src/main.mak
+++ b/contrib/src/main.mak
@@ -330,7 +330,8 @@ UPDATE_AUTOCONFIG = for dir in $(AUTOMAKE_DATA_DIRS); do \
 
 RECONF = mkdir -p -- $(PREFIX)/share/aclocal && \
 	cd $< && autoreconf -fiv $(ACLOCAL_AMFLAGS)
-CMAKE = cmake . -DCMAKE_TOOLCHAIN_FILE=$(abspath toolchain.cmake) \
+CMAKE = cmake . -DCMAKE_POLICY_VERSION_MINIMUM=3.5 \
+		-DCMAKE_TOOLCHAIN_FILE=$(abspath toolchain.cmake) \
 		-DCMAKE_INSTALL_PREFIX=$(PREFIX)
 
 #
diff --git a/contrib/src/websockets/rules.mak b/contrib/src/websockets/rules.mak
index 3069585..68830c7 100644
--- a/contrib/src/websockets/rules.mak
+++ b/contrib/src/websockets/rules.mak
@@ -29,6 +29,6 @@ endif
 DEPS_websockets = zlib $(DEPS_zlib)
 
 .websockets: websockets .zlib toolchain.cmake
-	cd $< && $(HOSTVARS) CFLAGS="$(CFLAGS) $(EX_ECFLAGS)" $(CMAKE) -DLWS_WITH_SSL=0 -DLWS_WITHOUT_TEST_PING=1 -DLWS_WITHOUT_TEST_SERVER_EXTPOLL=1
+	cd $< && $(HOSTVARS) CFLAGS="$(CFLAGS) $(EX_ECFLAGS)" $(CMAKE) -DLWS_WITH_SSL=0 -DLWS_WITHOUT_TESTAPPS=1 -DLWS_WITHOUT_TEST_PING=1 -DLWS_WITHOUT_TEST_SERVER_EXTPOLL=1
 	cd $< && $(MAKE) VERBOSE=1 install
 	touch $@
