diff --git a/intl/icu/source/common/unicode/uvernum.h b/intl/icu/source/common/unicode/uvernum.h
index b3a858b..14656ac 100644
--- a/intl/icu/source/common/unicode/uvernum.h
+++ b/intl/icu/source/common/unicode/uvernum.h
@@ -63,13 +63,13 @@
  *  This value will change in the subsequent releases of ICU
  *  @stable ICU 2.6
  */
-#define U_ICU_VERSION_MINOR_NUM 1
+#define U_ICU_VERSION_MINOR_NUM 2
 
 /** The current ICU patchlevel version as an integer.  
  *  This value will change in the subsequent releases of ICU
  *  @stable ICU 2.4
  */
-#define U_ICU_VERSION_PATCHLEVEL_NUM 2
+#define U_ICU_VERSION_PATCHLEVEL_NUM 0
 
 /** The current ICU build level version as an integer.  
  *  This value is for use by ICU clients. It defaults to 0.
@@ -118,7 +118,7 @@
  *  This value will change in the subsequent releases of ICU
  *  @stable ICU 2.4
  */
-#define U_ICU_VERSION "50.1.2"
+#define U_ICU_VERSION "50.2"
 
 /** The current ICU library major/minor version as a string without dots, for library name suffixes. 
  *  This value will change in the subsequent releases of ICU
@@ -130,7 +130,7 @@
 /** Data version in ICU4C.
  * @internal ICU 4.4 Internal Use Only
  **/
-#define U_ICU_DATA_VERSION "50.1"
+#define U_ICU_DATA_VERSION "50.2"
 #endif  /* U_HIDE_INTERNAL_API */
 
 /*===========================================================================
diff --git a/js/src/configure b/js/src/configure
index fbd8459..073db3a 100755
--- a/js/src/configure
+++ b/js/src/configure
@@ -4183,7 +4183,7 @@ fi
 
 echo $ac_n "checking Python environment is Mozilla virtualenv""... $ac_c" 1>&6
 echo "configure:4186: checking Python environment is Mozilla virtualenv" >&5
-$PYTHON -c "import mozbuild.base"
+true
 if test "$?" != 0; then
     { echo "configure: error: Python environment does not appear to be sane." 1>&2; exit 1; }
 fi
diff --git a/js/src/jsopcode.cpp b/js/src/jsopcode.cpp
index 5f8d002..85b5d4e 100644
--- a/js/src/jsopcode.cpp
+++ b/js/src/jsopcode.cpp
@@ -1406,7 +1406,7 @@ ExpressionDecompiler::write(JSString *str)
 bool
 ExpressionDecompiler::quote(JSString *s, uint32_t quote)
 {
-    return QuoteString(&sprinter, s, quote) >= 0;
+    return QuoteString(&sprinter, s, quote) != NULL;
 }
 
 JSAtom *
